lib/rpmte.cc

```diff
@@ -159,9 +159,18 @@ static int addTE(rpmte p, Header h, fnpyKey key, rpmRelocation * relocs)
     p->arch = headerGetAsString(h, RPMTAG_ARCH);
     p->os = headerGetAsString(h, RPMTAG_OS);
 
-    /* gpg-pubkey's dont have os or arch (sigh), for others they are required */
-    if (!rstreq(p->name, "gpg-pubkey") && (p->arch == NULL || p->os == NULL))
+    if (p->arch == NULL || p->os == NULL) {
+	if (p->type == TR_REMOVED && rstreq(p->name, "gpg-pubkey")) {
+	    rpmlog(RPMLOG_WARNING, "erasing gpg-pubkey packages is deprecated. use rpmkeys --delete\n");
+	} else {
+	    goto exit;
+	}
+    }
+
+    if (p->type != TR_REMOVED && rstreq(p->name, "gpg-pubkey")) {
+	rpmlog(RPMLOG_ERR, "installing gpg-pubkey packages is not allowed. use rpmkeys --import instead\n");
```

Comment:
Same issue with the period, comma, capitals as above. Pointing to rpmkeys in this case seems a bit dubious actually because rpmkeys wont import that package either. 

Another UI wisdom is that "allowed" tends to send the wrong kind of message, even if we think of it in those terms. Maybe just "public keys cannot be installed as packages"?

Reply:
Reworded this, kept the reference to rpmkeys but made it more clear that this is not for packages.